components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java

```diff
@@ -310,9 +310,9 @@ private SAMLSSOServiceProviderDTO createSAMLSSOServiceProviderDTO(SAMLSSOService
 
         serviceProviderDTO.setNameIDFormat(serviceProviderDO.getNameIDFormat());
 
-        if (serviceProviderDO.getAttributeConsumingServiceIndex() != null && !serviceProviderDO
-                .getAttributeConsumingServiceIndex().equals("")) {
+        if (StringUtils.isNotBlank(serviceProviderDO.getAttributeConsumingServiceIndex())) {
             serviceProviderDTO.setAttributeConsumingServiceIndex(serviceProviderDO.getAttributeConsumingServiceIndex());
+            serviceProviderDTO.setEnableAttributeProfile(true);
```

Comment:
I don't think this is the right fix to this issue.
Shouldn't the data object holding SAML related configs read from the persistence layer has this respective property being set.

Reply:
At the time of DTO to DO mapping happens EnableAttributeProfile property is set to true depending on the AttributeConsumingServiceIndex value. So shouldn't we follow the same approach when mapping DO to DTO?

Reply:
+1 in that case. so it means this respective property is not persisted in the database and populated based on AttributeConsumingServiceIndex. Is that right

Reply:
yeah, we have 2 option either modify DO or follow the same logic used in current implementation

Reply:
ok. agreed for the approach followed